test/SlowTests/Client/Counters/QueryOnCounters.cs

```diff
@@ -0,0 +1,651 @@
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced
+                        .RawQuery<CounterResult>("from users as u select counter(u, \"Downloads\")")
+                        .ToList();
+
+                    Assert.Equal(3, query.Count);
+                    Assert.Equal(100, query[0].Downloads);
+                    Assert.Equal(200, query[1].Downloads);
+                    Assert.Null(query[2].Downloads);
+                }
+            }
+        }
+
+        [Fact]
+        public void RawQuerySelectSingleCounterWithCounterAlias()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User { Name = "Jerry" }, "users/1-A");
+                    session.Store(new User { Name = "Bob" }, "users/2-A");
+                    session.Store(new User { Name = "Pigpen" }, "users/3-A");
+
+                    session.Advanced.Counters.Increment("users/1-A", "downloads", 100);
+                    session.Advanced.Counters.Increment("users/2-A", "downloads", 200);
+                    session.Advanced.Counters.Increment("users/3-A", "likes", 300);
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced
+                        .RawQuery<CounterResult2>("from users select counter(\"downloads\") as DownloadsCount")
+                        .ToList();
+
+                    Assert.Equal(3, query.Count);
+                    Assert.Equal(100, query[0].DownloadsCount);
+                    Assert.Equal(200, query[1].DownloadsCount);
+                    Assert.Null(query[2].DownloadsCount);
+                }
+            }
+        }
+
+        [Fact]
+        public void RawQuerySelectSingleCounterRawValues()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User { Name = "Jerry" }, "users/1-A");
+                    session.Store(new User { Name = "Bob" }, "users/2-A");
+                    session.Store(new User { Name = "Pigpen" }, "users/3-A");
+
+                    session.Advanced.Counters.Increment("users/1-A", "downloads", 100);
+                    session.Advanced.Counters.Increment("users/2-A", "downloads", 200);
+                    session.Advanced.Counters.Increment("users/3-A", "likes", 300);
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced
+                        .RawQuery<CounterResult3>("from users as u select counter(u, \"downloads\", true)")
```

Comment:
This should actually be: ` counter(u, \"downloads\", raw)`

Reply:
this will look something like: `counter(u, \"downloads\", \"raw\")`
Maybe we can do: `counterRaw(u, \"downloads\")`

Reply:
Yes